test/middleware/express.test.ts

```diff
@@ -183,154 +191,216 @@ xdescribe('Express Middleware tests', () => {
-                expect(mockResponse.locals?.timestamp).toBeInstanceOf('number');
                 // confirm that this is timestamp +/- 5 minutes of now.
                 const now: number = Date.now().valueOf();
-                const diff: number = Math.abs(now - (mockResponse.locals?.timestamp || 0));
-                expect(diff).toBeLessThan(5 * 60);
+                const diff: number = Math.abs(
+                    now - (mockResponse.locals?.graphqlGate.timestamp || 0)
+                );
+                expect(diff).toBeLessThan(5 * 60 * 1000);
             });
         });
 
         describe('Correctly limits requests', () => {
             describe('Allows requests', () => {
-                test('...a single request', () => {
+                test('...a single request', async () => {
                     // successful request calls next without any arguments.
-                    middleware(mockRequest as Request, mockResponse as Response, nextFunction);
+                    await middleware(
+                        mockRequest as Request,
+                        mockResponse as Response,
+                        nextFunction
+                    );
                     expect(nextFunction).toBeCalledTimes(1);
                     expect(nextFunction).toBeCalledWith();
                 });
 
-                test('Multiple valid requests at > 1 second intervals', () => {
+                test('Multiple valid requests at > 10 second intervals', async () => {
+                    const requests = [];
                     for (let i = 0; i < 3; i++) {
-                        const next: NextFunction = jest.fn();
-                        middleware(complexRequest as Request, mockResponse as Response, next);
-                        expect(next).toBeCalledTimes(1);
-                        expect(next).toBeCalledWith();
-
-                        // advance the timers by 1 second for the next request
-                        jest.advanceTimersByTime(1000);
+                        requests.push(
+                            middleware(
+                                complexRequest as Request,
+                                mockResponse as Response,
+                                nextFunction
+                            )
+                        );
+                        // advance the timers by 10 seconds for the next request
+                        jest.advanceTimersByTime(10000);
+                    }
+                    await Promise.all(requests);
+                    expect(nextFunction).toBeCalledTimes(3);
+                    for (let i = 1; i <= 3; i++) {
+                        expect(nextFunction).nthCalledWith(i);
                     }
                 });
 
-                test('Multiple valid requests at within one second', () => {
+                test('Multiple valid requests at within one second', async () => {
+                    const requests = new Array(5).fill(0);
```

Comment:
You're filling an array with zeros and then push requests into it, so the array sent to `Promise.all()` will be `[0,0,0,0,0,req, req, req]`